edge/cmd/edgecore/app/server.go

```diff
@@ -82,12 +83,18 @@ offering HTTP client capabilities to components of cloud to reach HTTP servers r
 				}
 			}
 
-			// get edge node local ip
-			if config.Modules.Edged.NodeIP == "" {
-				hostnameOverride := util.GetHostname()
-				localIP, _ := util.GetLocalIP(hostnameOverride)
-				config.Modules.Edged.NodeIP = localIP
+			// Get edge node local ip only when the custiomInterfaceName has been set.
+			// Defaults to the local IP from the default interface by the default config
+			if config.Modules.Edged.CustomInterfaceName != "" {
+				klog.Infof("Getting IP address by custom interface: %s", config.Modules.Edged.CustomInterfaceName)
+				ip, err := netutil.ChooseBindAddressForInterface(config.Modules.Edged.CustomInterfaceName)
+				if err != nil {
+					klog.Errorf("Failed to get IP address by custom interface: %s: %v", config.Modules.Edged.CustomInterfaceName, err)
+					os.Exit(1)
```

Comment:
```suggestion
					klog.Fatal("Failed to get IP address by custom interface: %s: %v", config.Modules.Edged.CustomInterfaceName, err)
```

Reply:
I think printing the stacktrace, doesn't help to debug this, so an error message and exiting with in failure state is better.

Reply:
Sounds good. :)